src/main/java/teammates/ui/output/SessionResultsData.java

```diff
@@ -85,20 +85,31 @@ public static SessionResultsData initForStudent(SessionResultsBundle bundle, Stu
             Map<String, List<ResponseOutput>> otherResponsesMap = new HashMap<>();
 
             if (questionDetails.isIndividualResponsesShownToStudents()) {
+                boolean canStudentSeeResponsesToSelf = question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER)
```

Comment:
I would like to ask if this is ok or is there is a better way to do this. The intention here is to check if self responses should be added to `responsesToSelf` array. Since the responses have already been filtered, I used this to check if the responses are supposed to be visible to any student or participant (they would have been filtered out beforehand if they are not supposed to be visible to this user).